modules/oe_theme_helper/src/Plugin/field_group/FieldGroupFormatter/FieldInPageNavigationFormatter.php

```diff
@@ -0,0 +1,188 @@
+<?php
+
+declare(strict_types = 1);
+
+namespace Drupal\oe_theme_helper\Plugin\field_group\FieldGroupFormatter;
+
+use Drupal\field_group\FieldGroupFormatterBase;
+
+/**
+ * Format a field group to create in-page navigation.
+ *
+ * @FieldGroupFormatter(
+ *   id = "oe_theme_helper_in_page_navigation",
+ *   label = @Translation("Field in-page navigation"),
+ *   description = @Translation("Format field group for ecl-inpage-navigation."),
+ *   supported_contexts = {
+ *     "view"
+ *   }
+ * )
+ */
+class FieldInPageNavigationFormatter extends FieldGroupFormatterBase {
+
+  /**
+   * {@inheritdoc}
+   */
+  public function process(&$element, $processed_object) {
+    $element += [
+      '#type' => 'field_group_in_page_navigation',
+    ];
+    // Get children elements from the in-page navigation group.
+    $field_elements = $processed_object['#fieldgroups'][$element['#group_name']]->children;
+    $links = [];
+    foreach ($field_elements as $value) {
+      // Check if we should add the field to page contents.
+      if (!is_array($element[$value]) || $this->excludeField($value, $element[$value], $processed_object['#fieldgroups']) === TRUE) {
+        continue;
+      }
+      // We will add anchors when the element, field or group, has a label.
+      $label = $this->fieldGetLabel($value, $processed_object);
```

Comment:
I like the idea of only displaying links if the field group item has a label or not, but let's just check if `#label_display` is `visible`. Also let's have a method only dedicated to that, call it `hasVisibleLabel(...): bool`.

Reply:
I implemented hasVisibleLabel(), the way visibility is controlled on fields and groups is different, so we need to control both cases.